src/Streamly/Internal/Data/Fold/Type.hs

```diff
@@ -704,12 +704,48 @@ serialWith func (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
 -- | Same as applicative '*>'. Run two folds serially one after the other
 -- discarding the result of the first.
 --
--- /Unimplemented/
---
+
+{-# ANN type SerialRunner Fuse #-}
+data SerialRunner sL sR
+    = SerialLeft !sL 
+    | SerialRight !sR
+
 {-# INLINE serial_ #-}
-serial_ :: -- Monad m =>
+serial_ :: Monad m =>
     Fold m x a -> Fold m x b -> Fold m x b
-serial_ _f1 _f2 = undefined
+serial_ (Fold stepL beginL _) (Fold stepR beginR doneR) =
```

Comment:
The behaviour of this is not similar to `serialWith`. This should be similar to `serialWith`. As a matter of fact, you can just copy the code from there, remove redundant stuff and use new constructors accordingly.

Reply:
As per the comment this  "Run two folds serially one after the other and discarding the result of the first."
It doesn't look similar to "serialWith". Let me know if I'm missing something here.

Reply:
It is a sepcial case of `serialWith`. It is in fact same as `serialWith (\_ b -> b)`. We are writing a separate implementation for this because this could be a little bit more efficient. It would be nice to have this code written similar to `serialWith` for comparison. It is in fact already similar, you can just compare the two and see if there are any differences that can be removed.